### PR TITLE
Add kvint4/8 ete testcase

### PR DESCRIPTION
**Pull request by zhulinJulia24**
1. benchmark.yaml change
- add kvint4/8 benchmark
- change cuda allocate for more effectiveness
2. daily_ete_test.yaml change
- remove kvint8 quantization
- remove timeout setting and reorder teststep because the testcase is much more stable
3. evaluate.yaml change
- change model path
4. pr_test_case chage 
- remove -s -v to make result more simpler and clearer
5. autotest/chat_prompt_case.yaml  and autotest/prompt_case.yaml
- react testcase to reduce error results
6. autotest/config.yaml and autotest/utils/config_utils.py
- split models into chat model and base model and add more models
- refactor kvint setting
7. autotest/interface/restful/test_restful_interface_func_common.py
- add more testcase into restful interface
8. autotest/tools/chat/*
- Streamline test cases, use one test case to cover multiple previous test cases
- split testcases into chat and base model 
9. autotest/tools/pipeline/*
- add kvint4/8 testcases
10. autotest/tools/restful/*
- add kvint4/8 testcases
- refactor start and stop restful server process

other changes:
- rename -inner-w4a16 models into -inner-4bits
- change temperater=0.01 into top_k=1 to make response more stable
- remove useless code after refactor kvint4/8 quantization

### CODE REVIEW

**Comment by zhulinJulia24** [comment]
## test result


### daily ete testcase:
https://github.com/zhulinJulia24/lmdeploy/actions/runs/8720843216
some response in some model is still not stable. I will still follow and improve it.
image


### benchmark testcase:
https://github.com/zhulinJulia24/lmdeploy/actions/runs/8724161035
image
image


### evaluation testcase:
https://github.com/zhulinJulia24/lmdeploy/actions/runs/8720159669
image

and 
https://github.com/zhulinJulia24/lmdeploy/actions/runs/8724058314
image

**Comment by zhulinJulia24** [comment]
pr_test time cost reduce to about 30mins, because of streamline test cases, use one test case to cover multiple previous test cases in cli chat. Currently 4bits quantization is still reserved.

File: .github/scripts/set_benchmark_param.sh [hunk 1]
```diff
@@ -5,7 +5,7 @@ then
     exit 1
 fi
 
-if [[ $1 == *"w4a16" ]] || [[ $1 == *"4bit" ]]
+if [[ $1 == *"w4a16" ]] || [[ $1 == *"4bit"* ]]
```
Comment (lvhan028) [inline .github/scripts/set_benchmark_param.sh:8]: 这里建议再把 *"awq" 加上。比如说，huggingface上现成的 4bit 模型：Qwen/Qwen1.5-7B-Chat-AWQ

File: .github/workflows/benchmark.yml [hunk 2]
```diff
@@ -159,16 +160,14 @@ jobs:
         run: |
           rm -rf ${result_dir}
           mkdir ${result_dir}
-          export CUDA_VISIBLE_DEVICES=6,7
           python3 benchmark/profile_generation.py $MODEL_PATH $MAX_ENTRY_COUNT $MODEL_FORMAT $TP_INFO -c 128 -ct 128 2048 128 -pt 128 128 2048 --csv ${result_dir}/generation.csv > ${result_dir}/generation.log
       - name: Run generation benchmark - pytorch
-        if: (! contains( matrix.model, 'w4a16') && !contains( matrix.model, '4bit')) && contains(fromJSON(github.event.inputs.backend), 'pytorch')
+        if: ( !contains( matrix.model, '4bit') && contains(fromJSON(github.event.inputs.backend), 'pytorch'))
```
Comment (lvhan028) [inline .github/workflows/benchmark.yml:165]: `!contains` 前面多了空格

File: .github/workflows/benchmark.yml [hunk 3]
```diff
@@ -388,16 +416,67 @@ jobs:
...
+          result_dir: benchmark-restful-turbomind-kvint4
+        run: |
+          rm -rf ${result_dir}
+          mkdir ${result_dir}
+          batches=(128 256)
+          for batch in "${batches[@]}"
+          do
+            for ((i=1; i<=$LOOP_NUM; i++))
+              do
+              python3 benchmark/profile_restful_api.py localhost:23333 $MODEL_PATH $DATASET_FILE --concurrency "$batch" --stream-output True --csv ${result_dir}/restful_csv_batch_"${batch}"_"${i}"th.csv &> ${result_dir}/restful_csv_batch_"${batch}"_"${i}"th.log
```
Comment (lvhan028) [inline .github/workflows/benchmark.yml:438]: 缩进一下比较好

File: .github/workflows/benchmark.yml [hunk 4]
```diff
@@ -388,16 +416,67 @@ jobs:
...
+          result_dir: benchmark-restful-turbomind-kvint8
+        run: |
+          rm -rf ${result_dir}
+          mkdir ${result_dir}
+          batches=(128 256)
+          for batch in "${batches[@]}"
+          do
+            for ((i=1; i<=$LOOP_NUM; i++))
+              do
+              python3 benchmark/profile_restful_api.py localhost:23333 $MODEL_PATH $DATASET_FILE --concurrency "$batch" --stream-output True --csv ${result_dir}/restful_csv_batch_"${batch}"_"${i}"th.csv &> ${result_dir}/restful_csv_batch_"${batch}"_"${i}"th.log
```
Comment (lvhan028) [inline .github/workflows/benchmark.yml:464]: 缩进一下比较好

File: .github/workflows/benchmark.yml [hunk 5]
```diff
@@ -388,16 +416,67 @@ jobs:
...
+              done
+          done
+      - name: Kill restful api turbomind - kvint8
+        continue-on-error: true
+        if: always()
+        run: |
+          kill -15 "$restful_pid"
       - name: Start restful api pytorch
-        if: (! contains( matrix.model, 'w4a16') && !contains( matrix.model, '4bit')) && contains(fromJSON(github.event.inputs.backend), 'pytorch')
+        if: ( !contains( matrix.model, '4bit') && contains(fromJSON(github.event.inputs.backend), 'pytorch'))
```
Comment (lvhan028) [inline .github/workflows/benchmark.yml:473]: `!contains` 前面多了空格

File: .github/workflows/daily_ete_test.yml [hunk 6]
```diff
@@ -139,36 +140,30 @@ jobs:
         run: |
           python3 -m pip install lmdeploy-*.whl
           python3 -m pip install -r requirements/test.txt
+          pip install /nvme/qa_test_models/offline_pkg/DeepSeek-VL
```
Comment (lvhan028) [inline .github/workflows/daily_ete_test.yml:143]: 加上选项 --no-deps

File: .github/workflows/daily_ete_test.yml [hunk 7]
```diff
@@ -139,36 +140,30 @@ jobs:
...
       - name: Install lmdeploy - offline
         if: ${{inputs.offline_mode}}
         run: |
           python3 -m pip install /nvme/qa_test_models/offline_pkg/py38/lmdeploy-*.whl
           python3 -m pip install -r requirements/test.txt
+          pip install /nvme/qa_test_models/offline_pkg/DeepSeek-VL
```
Comment (lvhan028) [inline .github/workflows/daily_ete_test.yml:149]: 加上选项 --no-deps

**Comment by lvhan028** [comment]
> pr_test time cost reduce to about 30mins, because of streamline test cases, use one test case to cover multiple previous test cases in cli chat. Currently 4bits quantization is still reserved.

How long did `pr_test` workflow cost before this PR?

**Comment by zhulinJulia24** [comment]
> > pr_test time cost reduce to about 30mins, because of streamline test cases, use one test case to cover multiple previous test cases in cli chat. Currently 4bits quantization is still reserved.
> 
> How long did `pr_test` workflow cost before this PR?

currently about 1hours.

**Comment by zhulinJulia24** [comment]
--no-deps fixed 

image

**Review by lvhan028 — APPROVED** [review]

**Review by RunningLeon — APPROVED** [review]
LGTM

**Pull request merged**